lib/pharos/ssh/client.rb

```diff
@@ -98,35 +103,35 @@ def exec_script!(name, env: {}, path: nil, **options)
         cmd.concat(EXPORT_ENVS.merge(env).map { |key, value| "#{key}=\"#{value}\"" })
         cmd.concat(%w(bash --norc --noprofile -x -s))
         logger.debug { "exec: #{cmd}" }
-        exec!(cmd, stdin: script, source: name, **options)
+        synchronize { exec!(cmd, stdin: script, source: name, **options) }
```

Comment:
These are wrapped twice with `synchronize`? Probably that's a bit too much?

Reply:
Yeah. Monitor implements a reentrant mutex though, but still unnecessary.